### DISCUSSION TITLE
Models Update

### DISCUSSION THREAD

**Discussion by PotatoPTT** [QIN2DIM/hcaptcha-challenger#171] [Q&A]
When a model is updated on GitHub and I have the old version downloaded for solving, it try to download the latest version or use the version that was previously downloaded?

**Comment by QIN2DIM** [comment]
This is a very difficult question to answer

**Comment by QIN2DIM** [comment]
对于这个问题，程序会尽可能帮你更换最新版本的模型。

我们使用了一个 `Assets`  Object 以及一系列延迟反射的机制确保分布式网络的两端一致性，同时尽最大可能做到“请求消峰”，减少对远端 ModelHub 的访问频次。

**Comment by QIN2DIM** [comment]
https://github.com/QIN2DIM/hcaptcha-challenger/blob/660b2ac2b3f6016c0428aade54c413a7b92a8d8f/src/services/hcaptcha_challenger/solutions/kernel.py#L235-L272
